src/main/java/com/redhat/devtools/lsp4ij/settings/jsonSchema/LSPServerConfigurationJsonSchemaFileProvider.java

```diff
@@ -0,0 +1,84 @@
+package com.redhat.devtools.lsp4ij.settings.jsonSchema;
+
+import com.intellij.openapi.application.ModalityState;
+import com.intellij.openapi.application.WriteAction;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.impl.PsiManagerEx;
+import com.intellij.psi.impl.file.impl.FileManagerImpl;
+import com.intellij.util.ModalityUiUtil;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.Objects;
+
+public class LSPServerConfigurationJsonSchemaFileProvider extends AbstractLSPJsonSchemaFileProvider {
+
+    private static final Object SCHEMA_CHANGE_REQUESTOR = new Object();
+
+    private final int index;
+    private final @NotNull Project project;
+    private MyLightVirtualFile file;
+    private boolean unused;
+
+    public LSPServerConfigurationJsonSchemaFileProvider(int index, @NotNull Project project) {
+        super(generateJsonFileName(index));
+        this.index = index;
+        this.project = project;
+        this.file = new MyLightVirtualFile("lsp.server." + index + ".schema.json", "");
+        this.unused = true;
+    }
+
+    private static String generateJsonFileName(int index) {
+        return "lsp.server." + index + ".json";
+    }
+
+    public int getIndex() {
+        return index;
+    }
+
+    @Override
+    public @Nullable VirtualFile getSchemaFile() {
+        return file;
+    }
+
+    public void reset() {
+        try {
+            updateFileContent("", file, project);
+        } finally {
+            unused = true;
+        }
+    }
+
+    public void setSchemaContent(@NotNull String jsonSchemaContent) {
+        try {
+            updateFileContent(jsonSchemaContent, file, project);
+        } finally {
+            unused = false;
+        }
+    }
+
+    public boolean isUnused() {
+        return unused;
+    }
+
+
+    private static void updateFileContent(@NotNull String content,
+                                          @NotNull MyLightVirtualFile file ,
+                                          @NotNull Project project) {
+        if (Objects.equals(content, file.getContent())) {
+            return;
+        }
+        file.setContent(SCHEMA_CHANGE_REQUESTOR, content, false);
+        reloadPsi(file, project);
+    }
+
+    private static void reloadPsi(final VirtualFile file, final Project project) {
+        final FileManagerImpl fileManager = (FileManagerImpl) PsiManagerEx.getInstanceEx(project).getFileManager();
+        if (fileManager.findCachedViewProvider(file) != null) {
+            ModalityUiUtil.invokeLaterIfNeeded(ModalityState.defaultModalityState(), project.getDisposed(),
+                    () -> WriteAction.run(() -> fileManager.forceReload(file))

```

Comment:
Is this your proposed alternative to `refreshBundledFile()` to force VFS to refresh? Have you confirmed that it works for physical bundled virtual files?